### DISCUSSION TITLE
One to Many Relationship filtering

### DISCUSSION THREAD

**Discussion by RyanGaudion** [supabase/supabase#9332] [Questions] [answered]
Hi, I have 2 tables: shared_goals & shared_goals_values with a 1 to many relationship. I'm using this query to get them:

```
const {data:goals, error} = await supabase.from('shared_goals').select(`id, title, displayType, target, shared_goals_values(email, value)`)
```

I would like to filter (via the database query) if shared_goals_values[].email contains a certain email.

If I run this:

```
const {data:goals, error} = await supabase.from('shared_goals').select(`*, shared_goals_values!inner(*)`).eq('shared_goals_values.email', 'test@test.com')
```
It only returns shared_goals_values where email equals the entered email address. Instead, I would like to return all  shared_goals_values if shared_goals.shared_goals_values contains the selected email.

For instance there may be 2 shared_goals_values for a single shared_goals. If one of the shared_goals_values contains "test@test.com" I would like to return the full shared_goals object however if neither of them match then don't return the shared_goals at all.

Hope this makes sense

**Comment by RyanGaudion** [comment, marked as answer]
The command below is working, however returning array "ignore" seems bad. What should this command be?

```
const {data:goals, error} = await supabase.from('shared_goals').select(`id, ignore:shared_goals_values!inner(*), shared_goals_values(*)`).eq('ignore.email', 'rgaudion@gmail.com')
```

**Reply by steve-chavez** [reply]
For now you'll have to ignore the `ignore` array client-side.

There's an enhancement planned that will allow you to have an empty parentheses `shared_goals_values()` to remove the result server-side.

https://github.com/PostgREST/postgrest/issues/2340

**Reply by RyanGaudion** [reply]
@steve-chavez thanks for the reply. I'll keep an eye on that PostREST issue.
